unit_tests/test_socket_connection.py

```diff
@@ -259,6 +259,8 @@ class TestSocketConnection(unittest.TestCase):
     hardware or network dependent.
     """
 
+    # TODO: Remove pytype ignore when SocketConnection is removed
+    # pytype: disable=attribute-error
```

Comment:
Why does SocketConnection mean we need the ignore?

Reply:
Since SocketConnection is deprecated it only passes the arguments to the appropriate class based on BaseSocketConnection.
Pytype raises following error without the attribute-error ignore:
```python
test_socket_connection.py", line 285, in test_tcp_client: No attribute 'open' on None [attribute-error]
  In Optional[Union[boofuzz.connections.raw_l2_socket_connection.RawL2SocketConnection, boofuzz.connections.raw_l3_socket_connection.RawL3SocketConnection, boofuzz.connections.tcp_socket_connection.TCPSocketConnection, boofuzz.connections.udp_socket_connection.UDPSocketConnection]]
```
I can only guess, but I think this is because of the `raise` statements in SocketConnection which would lead to a None return.

The error doesn't appear when referring to the new socket classes directly, hence the TODO.